### PR TITLE
Adding option to read in JSON resulting from MaIS API by adding a value to Setting

### PR DESCRIPTION
**Pull request by hudajkhan**
Previous pull request, starting with https://github.com/sul-dlss/course_reserves/pull/474, helped setup a rake task that could be run to generate JSON files with course information from the MaIS APIs.

This pull request allows the app to read in those JSON files if a particular Setting key "use_course_json" is present.  The default is to not have that variable be in the Settings file, so we can try out how the app behaves in staging first by manually adding that setting.  The tests have been updated to pretend the use_course_json setting is present. 

A subsequent pull request will do the following to switch the course reserves process over to using the MaIS APIs:
- Update schedule.rb so it runs the fetch_courses rake task instead of fetch_xml 
- Remove the setting based option to use the MaIS API JSON and switch over entirely (and remove the registry harvester XML pathway)
- Update the tests so they no longer rely on the "use_course_json" setting (since they will no longer have to)

How to test this pull request:
- There are files on staging with the resulting JSON from the APIs (under the lib/course_work_content directory).  If you copy these over to your local lib/course_work_content directory, then start the app as you normally would, and then click on "create new reserve list", you should see several pages of courses listed.

### CODE REVIEW

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: lib/course_work_courses.rb [hunk 1]
```diff
@@ -212,14 +212,15 @@ def process_all_courses(json_files)
     # and create the objects required by the model
     json_files.each do |json_file|
       json_file.each do |course|
-        if course.key?("instructors") && !course["instructors"].empty?
+        course = course.with_indifferent_access
+        if course.key?(:instructors) && !course[:instructors].empty?
```
[the remark this answers is not in the document]
Comment (hudajkhan) [reply lib/course_work_courses.rb]: Changed this as suggested

File: lib/course_work_courses.rb [hunk 2]
```diff
@@ -212,14 +212,15 @@ def process_all_courses(json_files)
...
-            cid: course["cid"],
-            cids: course["cids"],
-            sid: course["sid"],
-            instructors: course["instructors"].map { |inst| inst.transform_keys(&:to_sym) }
+            title: course[:title],
+            term: course[:term],
+            cid: course[:cid],
+            cids: course[:cids],
+            sid: course[:sid],
+            instructors: course[:instructors]
```
[the remark this answers is not in the document]
Comment (hudajkhan) [reply lib/course_work_courses.rb]: The course object from what is being read in already has the same fields as required by the Course Ruby initializer.  Additionally, for this to work, all the hash keys have to be symbols so I had to change that using deep_transform_keys (b/c instructors is a nested hash).  This section now reads as:

`course.deep_transform_keys!(&:to_sym)
  yield Course.new(**course) if course[:instructors].present?`

File: lib/course_work_courses.rb [hunk 3]
```diff
@@ -50,6 +60,19 @@ def raw_xml
     @raw_xml ||= self.raw_xml
   end
 
+  # Handling JSON
+  def initialize_json(json_file = nil)
```
[the remark this answers is not in the document]
Comment (hudajkhan) [reply lib/course_work_courses.rb]: Updated this

File: lib/course_work_courses.rb [hunk 4]
```diff
@@ -38,7 +38,17 @@ def self.instance
...
+    if Settings.use_course_json
+      initialize_json(file)
+    else
+      initialize_xml(file)
+    end
+  end
+
+  # These two initialize methods are broken out to enable both approaches
+  # until we switch over completely to JSON
+  def initialize_xml(xml = nil)
```
[the remark this answers is not in the document]
Comment (hudajkhan) [reply lib/course_work_courses.rb]: Updated this

**Review by jcoyne — APPROVED** [review]

**Pull request merged**